tensorflow/python/ops/summary_ops_v2.py

```diff
@@ -1226,8 +1225,12 @@ def trace_export(name, step=None, profiler_outdir=None):
     if _current_trace_context is None:
       raise ValueError("Must enable trace before export.")
     graph, profiler = _current_trace_context  # pylint: disable=redefined-outer-name
+    if profiler_outdir is None \
+        and isinstance(_summary_state.writer, ResourceSummaryWriter):
+      profiler_outdir = _summary_state.writer._init_op_fn.keywords.get('logdir')  # pylint: disable=protected-access
```

Comment:
This is a bit too much of a hack, I think - it would be easy to break this accidentally.

To do this explicitly, let's instead change ResourceSummaryWriter so that it takes an additional keyword argument `metadata` which becomes a `_metadata` property on the writer, as a container for arbitrary metadata about the writer.  The arg should default to None (which turns into the empty dictionary) and in the logdir-based writer case we can pass a dictionary `{"logdir": logdir}` as that argument.

Then here we can do

```python
if (profiler_outdir is None
    and isinstance(_summary_state.writer, ResourceSummaryWriter)):
  logdir = _summary_state.writer._metadata.get("logdir")  # pylint: disable=protected-access
  if logdir is not None:
    profiler_outdir = logdir
```

Reply:
Thank you! Your point is right!
I added `_metadata` property to `ResourceSummaryWriter` and used it !